opentelemetry-api/src/opentelemetry/trace/span.py

```diff
@@ -117,6 +117,23 @@ def add_event(
         timestamp if the `timestamp` argument is omitted.
         """
 
+    def add_link(  # pylint: disable=no-self-use
+        self,
+        context: "SpanContext",
+        attributes: types.Attributes = None,
+    ) -> None:
+        """Adds a `Link`.
+
+        Adds a single `Link` with the `SpanContext` of the span to link to and,
+        optionally, attributes passed as arguments. Implementations may ignore
+        calls with an invalid span context.
+
+        Note: It is preferred to add links at span creation, instead of calling
+        this method later since samplers can only consider information already
+        present during span creation.
+        """
+        _logger.warning("Span.add_link() is a no-op.")
```

Comment:
Can you make the message more actionable? Something like "Span.add_link() not implemented and will be a no-op. Use opentelemetry-sdk >= x.y to add links after span creation".

Using `_logger` would cause log spam every time this is executed. I think the usual way to do a one-time warning is `warnings.warn()`. @ocelotl I remember we used this previously and it also caused spam, can you advise the best way to accomplish a one-time warning per process?

Reply:
I believe warnings should only be emitted once:

> [Repetitions of a particular warning for the same source location are typically suppressed.](https://docs.python.org/3/library/warnings.html#:~:text=Repetitions%20of%20a%20particular%20warning%20for%20the%20same%20source%20location%20are%20typically%20suppressed.)

Reply:
@aabmass @ocelotl I've made the required changes. Can you have a look?